base/linalg/adjtrans.jl

```diff
@@ -0,0 +1,120 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+using Base: @pure, @propagate_inbounds, _return_type, _default_type, _isleaftype, @_inline_meta
+import Base: length, size, indices, IndexStyle, getindex, setindex!, parent, vec, convert, similar
+
+### basic definitions (types, aliases, constructors, abstractarray interface, sundry similar)
+
+# note that Adjoint and Transpose must be able to wrap not only vectors and matrices
+# but also factorizations, rotations, and other linear algebra objects, including
+# user-defined such objects. so do not restrict the wrapped type.
+struct Adjoint{T,S} <: AbstractMatrix{T}
+    parent::S
+    function Adjoint{T,S}(A::S) where {T,S}
+        checkeltype(Adjoint, T, eltype(A))
+        new(A)
+    end
+end
+struct Transpose{T,S} <: AbstractMatrix{T}
+    parent::S
+    function Transpose{T,S}(A::S) where {T,S}
+        checkeltype(Transpose, T, eltype(A))
+        new(A)
+    end
+end
+
+@pure function checkeltype(::Type{Transform}, ::Type{ResultEltype}, ::Type{ParentEltype}) where {Transform, ResultEltype, ParentEltype}
```

Comment:
Why does this need to be `@pure`

Reply:
This definition was preexisting for `RowVector`s :).